sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+
+        self.write_output()
+
+    def initalise_scenario(self: ConfigurationOutput, scenario_file: Path) -> None:
+        """Reads scenario file and initalises ConfigurationScenario."""
+        config = {}
+        with scenario_file.open() as file:
+            for line in file:
+                key, value = line.strip().split(" = ")
+                config[key] = value
+
+        instance_set_train = resolve_object_name(
+            config["instance_file"],
+            gv.file_storage_data_mapping[gv.instances_nickname_path],
+            gv.instance_dir, InstanceSet)
+        # TODO: Find out why _train.txt is added to end of name
+        # Remove .txt to not cause any issues with later use of
+        # name in self.configurator.scenario._set_paths()
+        instance_set_train.name = instance_set_train.name.replace("_train.txt", "")
+
+        # Set instance_set_test to later retrieve validation results
+        instance_set_test = resolve_object_name(
+            config["test_instance_file"],
+            gv.file_storage_data_mapping[gv.instances_nickname_path],
+            gv.instance_dir, InstanceSet)
+        if instance_set_test is not None:
+            self.instance_set_test = instance_set_test
+
+        # Collect relevant settings
+        if "cpu_time" in config:
+            cpu_time = int(config["cpu_time"])
+        else:
+            cpu_time = None
+
+        if "wallclock-limit" in config:
+            wallclock_limit = int(config["wallclock-limit"])
+        else:
+            wallclock_limit = None
+
+        if "runcount-limit" in config:
+            solver_calls = int(config["runcount-limit"])
+        else:
+            solver_calls = None
+
+        if "feature_file" in config:
+            use_features = bool(config["feature_file"])
+        else:
+            use_features = None
+
+        # TODO: Add METRIC to objective
+        objective = SparkleObjective(f"{config['run_obj']}:UNKNOWN")
+
+        # TODO: Number_of_runs isn't part of the scenario file
+        self.configurator.scenario = ConfigurationScenario(self.solver,
+                                                           instance_set_train,
+                                                           None,
+                                                           solver_calls,
+                                                           cpu_time,
+                                                           wallclock_limit,
+                                                           int(config["cutoffTime"]),
+                                                           config["cutoff_length"],
+                                                           objective,
+                                                           use_features)
+        self.configurator.scenario._set_paths(self.configurator.output_path)
+
+    def get_validation_data(self: ConfigurationOutput, instance_set: InstanceSet) -> \
```

Comment:
The method could be a bit cleaner, but the structure is as I would have expected 👍

Reply:
Any specifics that you would like to see changed?
I though about grouping the creation of the ValidationResults for both config and default through a loop or something, however I decided that it probably would be easier to understand if I didn't do that.